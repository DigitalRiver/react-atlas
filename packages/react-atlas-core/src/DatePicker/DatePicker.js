import React from "react";
import PropTypes from "prop-types";
import cx from "classnames";
import { default as Dp } from "react-datepicker";
import moment from "moment";

class DatePicker extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      "startDate": moment(this.props.selected) || moment()
    };
    this.handleChange = this.handleChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (
      typeof nextProps.selected !== "undefined" &&
      nextProps.selected !== this.state.selected
    ) {
      this.setState({ "startDate": moment(nextProps.selected) });
    }
  }

  handleChange(date, event) {
    event.persist();
    this.setState({
      "startDate": date
    });
    if (typeof this.props.onChange !== "undefined") {
      this.props.onChange(date, event, true);
    }
  }

  render() {
    let { format, name, className, style } = this.props;
    return (
      <Dp
        style={style}
        className={cx(className, "ra_DatePicker__datepicker")}
        name={name}
        dateFormat={format}
        selected={this.state.startDate}
        onChange={this.handleChange}
      />
    );
  }
}

DatePicker.propTypes = {
  /**
   * An object, array, or string of CSS classes to apply to DatePicker.
   */
  "className": PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array
  ]),

  /**
   * A date format that will be applied to the displayed date.  Default is MM/DD/YYYY.
   */
  "format": PropTypes.string,

  /**
   * The name that will be applied to the input element.
   */
  "name": PropTypes.string,

  /**
   *  The function that will be executed on change.
   */
  "onChange": PropTypes.func,

  /**
   * The currently selected Date.
   */
  "selected": PropTypes.string,

  /**
   * Pass inline styles here.
   */
  "style": PropTypes.object
};

export default DatePicker;
