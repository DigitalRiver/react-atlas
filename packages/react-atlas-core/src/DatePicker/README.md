

Date picker:

      <DatePicker onChange={function(date) {console.log("date: ", date._d)}}/>
