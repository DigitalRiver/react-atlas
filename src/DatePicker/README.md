

DatePicker with onChange:

      <DatePicker onChange={function(date) {console.log("date: ", date._d)}}/>

DatePicker with format:

      <DatePicker format="MMMM Do YYYY"/>


DatePicker with selected prop and selected date:

		initialState = {
			date: "11/11/2011" 
		}

		handleDateChange = () => {
			setState({ "date": "12/12/2012"})
		};

		<div>
			<DatePicker selected={state.date} onChange={function(date) {console.log("date: ", date._d)}}/>
			<Button onClick={handleDateChange}> Change </Button>
			<br />
			State date: {state.date.toString()}
		</div>
