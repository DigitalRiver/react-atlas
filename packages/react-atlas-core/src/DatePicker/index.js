import { default as DatePickerCore } from "./DatePicker";
export { DatePickerCore };
