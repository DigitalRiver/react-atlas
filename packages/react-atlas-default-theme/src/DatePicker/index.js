import * as DatePickerStyle from "./DatePicker.css";
export { DatePickerStyle };
