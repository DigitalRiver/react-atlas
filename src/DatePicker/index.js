import DatePicker from "./DatePicker.js";
export { DatePicker };
