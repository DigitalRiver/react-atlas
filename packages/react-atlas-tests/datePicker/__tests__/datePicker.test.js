import React from "react";
import { mount, shallow } from "enzyme";
import { DatePickerCore } from "../../../react-atlas-core/src/DatePicker/index";
import moment from "moment";

import renderer from "react-test-renderer";

describe("DatePicker Test suite", () => {
  it("DatePicker - Basic test", function() {
    const dp = mount(<DatePickerCore />);
  });

  it("DatePicker - Basic test (selected)", function() {
    const dp = mount(<DatePickerCore selected={"2017-11-29"} />);
  });

  it("DatePicker - Basic test fixed date", function() {
    const dp = mount(
      <DatePickerCore
        onChange={function(date) {
          console.log("date: ", date._d);
        }}
      />
    );
  });

  it("DatePicker - change & refresh component", function() {
    const dp = mount(<DatePickerCore selected={"2017-11-29"} />);
    dp.update();
  });

  it("DatePicker - Correct render test", function() {
    const tree = renderer
      .create(<DatePickerCore selected={"2017-11-29"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  function _simulateEnterText(input, str) {
    for (let i = 0; i < str.length; i++) {
      input.simulate("keydown", { which: str[i] });
    }
  }

  /*	This test is commented out until the "event not defined" issue is solved.
  it("DatePicker - Change event test ", function() {
    const dp = shallow(<DatePickerCore selected={"2017-11-29"} onChange={(date)=>console.log("[INFO] onChange function triggered")}/>);
		let expectedValue = moment("2017-11-29");
		
		expect(dp.state().startDate.isSame(expectedValue)).toBe(true);
		
		let newValue = moment();
		dp.simulate('change', newValue);
		expect(dp.state().startDate.isSame(newValue)).toBe(true);
  });
	*/
});
