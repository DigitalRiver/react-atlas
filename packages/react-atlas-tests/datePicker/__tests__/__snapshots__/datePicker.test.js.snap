// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Test suite DatePicker - Correct render test 1`] = `
<div>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        autoComplete={undefined}
        autoFocus={undefined}
        className="ra_DatePicker__datepicker"
        disabled={false}
        id={undefined}
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder={undefined}
        readOnly={undefined}
        required={undefined}
        tabIndex={undefined}
        title={undefined}
        type="text"
        value="11/29/2017"
      />
    </div>
  </div>
</div>
`;
